dandi/tests/test_metadata.py

```diff
@@ -48,6 +50,29 @@ def test_get_metadata(simple1_nwb: str, simple1_nwb_metadata: Dict[str, Any]) ->
     assert target_metadata == metadata
 
 
+def test_bids_nwb_metadata_integration(bids_examples, tmp_path):
+    """
+    Notes
+    -----
+    * Generating data manually here, as fixture workflow calls `new_dandiset`,
+        which requires spinning up docker:
+        https://github.com/dandi/dandi-cli/pull/1183#discussion_r1061622910
+    """
+
+    source_dpath = os.path.join(bids_examples, "ieeg_epilepsyNWB")
+    dpath = os.path.join(tmp_path, "ieeg_epilepsyNWB")
+    shutil.copytree(source_dpath, dpath)
+    with open("example.txt", "w") as f:
+        f.write("identifier: '000001'")
+        f.close
+
+    file_path = os.path.join(
+        dpath, "sub-01/ses-postimp/ieeg/sub-01_ses-postimp_task-seizure_run-01_ieeg.nwb"
+    )
+    metadata = get_metadata(file_path)
+    assert metadata["subject_id"] == "01"
```

Comment:
is there any nwb specific metadata you could test here for to ensure that we get both nwb and bids metadata?

Reply:
added NWB specific key and key-comments in [de9061a](https://github.com/dandi/dandi-cli/pull/1183/commits/de9061aef6fdfb32ccb44359e3753509ef25f6a7)